### PR TITLE
Lets you emag boogiebots

### PR DESCRIPTION
**Pull request by BatElite**
[feat]
## About the PR 
This PR allows boogiebots to be emagged. An emagged boogiebot will make bystanders dance along with it when it starts dancing.

## Why's this needed? 
More silly emag interactions and more dancing are both good things.

## Changelog

```
(u)BatElite:
(+)You can now emag boogiebots!
```

### CODE REVIEW

**Comment by BatElite** [comment]
I'm not sure about the changelog. I want to make sure people know about it but I also don't want to spoil players what happens, so we're left with kinda nothing.

Don't feel strongly about keeping it, but otherwise I suspect nobody will ever find out it exists.

File: code/obj/critter/pets_small_animals.dm [hunk 1]
```diff
@@ -1793,6 +1794,13 @@ var/list/shiba_names = list("Maru", "Coco", "Foxtrot", "Nectarine", "Moose", "Pe
 				src.visible_message("<b>[src]</b> [msg]!",2)
 			src.icon_state = pick("boogie-d1","boogie-d2","boogie-d3")
 			// maybe later make it ambient play a short chiptune here later or at least some new sound effect
+			if (emagged)
+				SPAWN_DBG(0.5 SECONDS)
+					LAGCHECK(LAG_MED)
+					for (var/mob/living/carbon/human/responseMonkey in orange(2, src)) // they don't have to be monkeys, but it's signifying monkey code
```
Comment (ZeWaka) [inline code/obj/critter/pets_small_animals.dm:1800]: LAGCHECKS go inside for loops

File: code/obj/critter/pets_small_animals.dm [hunk 2]
```diff
@@ -1793,6 +1794,13 @@ var/list/shiba_names = list("Maru", "Coco", "Foxtrot", "Nectarine", "Moose", "Pe
...
+						if (responseMonkey.stat || responseMonkey.getStatusDuration("paralysis") || responseMonkey.sleeping || responseMonkey.getStatusDuration("stunned"))
```
Comment (ZeWaka) [inline code/obj/critter/pets_small_animals.dm:1801]: .stat checks is dumb - there's a wrapper proc for this whole chain

**Review by pali6 — APPROVED** [review]

File: code/obj/critter/pets_small_animals.dm [hunk 3]
```diff
@@ -1827,6 +1835,15 @@ var/list/shiba_names = list("Maru", "Coco", "Foxtrot", "Nectarine", "Moose", "Pe
 		else
 			. = ..()
 
+	emag_act(mob/user, obj/item/card/emag/E)
+		if (!src.emagged)
+			if(user)
+				boutput(user, "<span class='alert'>You short out the [src]'s dancing intensity setting to 'flashmob'.</span>")
+			src.visible_message("<span class='alert'><b>[src] lights up with determination!</b></span>")
+			src.emagged = 1
```
Comment (ZeWaka) [inline code/obj/critter/pets_small_animals.dm:1843]:
```suggestion
			src.emagged = TRUE
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**